### PR TITLE
First pass at schemas

### PR DESCRIPTION
**Pull request by kesne**
My space bar isn't working consistently.

@Emily 

So this is my first pass at schemas. Example usage in the example file.

My approach was to try to align somewhat with ActiveRecord. I had to inverse it from `t.string(key)` to `key: t.string`, otherwise it would not be possible to statically type.

My alternative idea was a big function chain, something like this:

```ts
const schema = new Schema(20080906171750)
  .createTable('users', { force: true }, table =>
    table
      .string('firstName', { optional: true })
      .string('lastName')
      .boolean('deleted')
      .datetime('createdAt')
      .datetime('updatedAt'),
  );
```

I thought that was less ideal than the proposed syntax though.

### CODE REVIEW

**Comment by kesne** [comment]
Also I know migrations will be similar and I have no idea what those will look like, so just let me know if this won't work.

**Comment by Emily** [comment]
The example you have is relatively simple. I think the alternative chained syntax actually works really well for more complicated things, like:

```typescript
const schema = new Schema(20080906171750)
  .createTable('users', { force: true }, table =>
    table
      .string('firstName', { optional: true })
      .string('lastName', { length: 400 })
      .string('email', { index: { unique: true, calculated: 'lower(email)' } })
      .boolean('deleted')
      .datetime('createdAt', { index: true })
      .datetime('updatedAt', { index: 'unique' }),
  );
```

But generally I think this syntax is just a subset of the migration syntax, and is directionally good.

**Comment by kesne** [comment]
Yeah the chained syntax is good for those cases, but it makes typing things like optional challenging. I'm not super sure how to make the optional type work for these cases. A functional call is all that I can figure out.

**Comment by kesne** [comment]
So I think the other option is to split the type def:

```ts
const schema = new Schema(20080906171750)
  .createTable('users', { force: true }, t => ({
    firstName: { type: t.optional(t.string) },
    lastName: { type: t.string, length: 400 },
    email: { type: t.string, index: { unique: true, calculated: 'lower(email)' } },
  });
```

I think optional needs to get coupled with the type def otherwise I don't know how we can type it correctly.

**Comment by kesne** [comment]
@Emily Added my second pass which turns everything into a function call + backing instance so that we can actually attach functionality and options. Still need to do nonNull and add other types, but this should do for now.

File: packages/fewer/src/example.ts [hunk 1]
```diff
@@ -1,14 +1,29 @@
...
-export const DeletedUsers = Users.where({ deleted: [true, true] }).where({deleted: false});
+const schema = new Schema(20080906171750)
+  .createTable('users', { force: true }, t => ({
+    firstName: t.optional(t.string),
+    lastName: t.string,
+    deleted: t.boolean,
+    createdAt: t.datetime,
+    updatedAt: t.datetime,
+  }))
+  .createTable('products', { force: true }, t => ({
```
Comment (Emily) [inline packages/fewer/src/example.ts]: This style of immutable chaining seems to be a common pattern. I could also see people getting confused by it, so we should probably call it out in documentation.

**Pull request merged**